sycl/include/CL/sycl/queue.hpp

```diff
@@ -267,14 +265,14 @@ class __SYCL_EXPORT queue {
         }
       };
 
-      Event = submit_impl_and_postprocess(CGF, CodeLoc, PostProcess);
+      auto Event = submit_impl_and_postprocess(CGF, CodeLoc, PostProcess);
+      return discard_or_return(Event);
     } else
 #endif // __SYCL_USE_FALLBACK_ASSERT
     {
-      Event = submit_impl(CGF, CodeLoc);
+      auto Event = submit_impl(CGF, CodeLoc);
+      return discard_or_return(Event);
```

Comment:
what is the reason to move the return statement in if-else blocks?
On first look, it causes code duplication and potential warnings from code sanitizers about missing return statements.

Reply:
The reason for this is to reduce the call to the move assignment operator sycl::event by one.